### PR TITLE
RA-1887, add showLegend and hideConcepts config parameters

### PR DESCRIPTION
**Pull request by cioan**
@mogoodrich , I have added two new config parameters to the obsgraph in order to satisfy the requirements for UHM-1676. 
I also included a small change @brandones asked regarding the maxAge parameter. Thanks!

### CODE REVIEW

File: omod/src/main/web/dashboardwidgets/dashboardwidgetscommons.service.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ export default class WidgetsCommons {
             today.setMonth(today.getMonth()-parseInt(maxAge));
         } else if( maxAge.indexOf('y') !== -1 ){
             maxAge = maxAge.replace('y', '');
-            today.setDate(today.getYear()-(parseInt(maxAge)));
+            today.setFullYear(today.getFullYear()-(parseInt(maxAge)));
```
Comment (mogoodrich) [inline omod/src/main/web/dashboardwidgets/dashboardwidgetscommons.service.js:84]: Just for reference, what does this change do @cioan  @brandones ?

File: omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js [hunk 2]
```diff
@@ -140,6 +144,7 @@ export default class ObsGraphController {
                           tempData.push(yValue);
                         }
                         this.data.push(tempData);
+                        console.log("updateChartData:tempData: " + JSON.stringify(tempData));
```
Comment (mogoodrich) [inline omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js:147]: stray debugging statement... :)

File: omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js [hunk 3]
```diff
@@ -124,6 +124,10 @@ export default class ObsGraphController {
             if (xAxisKeys !== null && xAxisKeys.length > 0) {
               for (let j=0; j < this.conceptArray.length; j++) {
                 let concept = this.conceptArray[j];
+                if( this.hideConcepts !== undefined && this.hideConcepts.length && this.hideConcepts.includes(concept.uuid)) {
+                  //if it was indicated that this concept should not be plotted, than skip to the next concept
+                  continue;
```
Comment (mogoodrich) [inline omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js:129]: I'm probably missing something obvious :) but if you don't want to graph a concept, why would you include it in the concept list in the first place?

**Review by mogoodrich — COMMENTED** [review]
Thanks @cioan !  Just had a couple questions and a stray debug line to remove...

File: omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js (hunk 3 above)
Comment (cioan) [reply omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js:129]: So I could get the values I need to calculate the BMI. I need the Weight obs to calculate the BMI but we do not want to plot the Weight line.

Comment (mogoodrich) [reply omod/src/main/web/dashboardwidgets/obsgraph/obsgraph.controller.js:129]: Ah, makes sense!

File: omod/src/main/web/dashboardwidgets/dashboardwidgetscommons.service.js (hunk 1 above)
Comment (cioan) [reply omod/src/main/web/dashboardwidgets/dashboardwidgetscommons.service.js]: Not sure, I just reverted this change to maxAge.

**Pull request merged**